src/ensembl/src/content/app/species-selector/components/species-selector-app-bar/SpeciesSelectorAppBar.test.tsx

```diff
@@ -42,46 +45,61 @@ jest.mock(
   () => () => <div>ConversationIcon</div>
 );
 
-const defaultProps = {
-  selectedSpecies: times(5, () => createSelectedSpecies()),
-  push
+const selectedSpecies = times(5, () => createSelectedSpecies());
+
+const defaultReduxState = {
+  speciesSelector: {
+    committedItems: selectedSpecies
+  }
+};
+
+const mockStore = configureMockStore([thunk]);
+
+const renderComponent = (state?: any) => {
+  state = Object.assign({}, defaultReduxState, state);
```

Comment:
`merge` or `Object.assign` better?  Both gives same result, so is it just a personal coding preference?

Reply:
> Both gives same result

They don't. Lodash `merge` does a deep merge. Object.assign does a shallow merge.

Example: given

```
const a = {
  foo: 1,
  bar: { x: 1 },
  baz: [ 1, 2 ]
};

const b = {
  foo: 2,
  bar: { y: 2 },
  baz: []
}
```

then `merge({}, a, b)` will produce:

```
{
  foo: 2,
  bar: { x: 1, y: 2 },
  baz: [ 1, 2 ]
}
```

while `Object.assign({}, a, b)` will produce:

```
{
  foo: 2,
  bar: { y: 2 },
  baz: []
}
```

P.S.: both lodash's `merge` and Object.assign mutate the first object that they receive. `merge` from `lodash/fp` will produce a new object; but we are rarely using `lodash/fp` (mostly just for the `set` function).